`test_pageserver_http_get_wal_receiver_success` is flaky

E.g. see here: https://app.circleci.com/pipelines/github/neondatabase/neon/6589/workflows/8ef21c54-1f2e-4346-98e3-accb4f4f0910/jobs/66732

```
test_runner/batch_others/test_pageserver_api.py:87: in test_pageserver_http_get_wal_receiver_success
    assert res2["last_received_msg_lsn"] > res["last_received_msg_lsn"]
E   TypeError: '>' not supported between instances of 'str' and 'NoneType'
```

My suspicion is that there is no guarantee that a timeline is fully(?) propagated to Pageserver once `env.postgres.create_start` is completed, so it's possible that `env.pageserver.http_client().wal_receiver_get(....)` returns `None` in the `last_received_msg_lsn` field.

If it's the case, who does the propagation? Is it just delayed by local network or is it done by some background thread on timeout? Should we wait for it instead? Should we wait in the `create_start` function itself?

--- a/test_runner/batch_others/test_pageserver_api.py
+++ b/test_runner/batch_others/test_pageserver_api.py
@@ -90,7 +90,7 @@ def test_pageserver_http_get_wal_receiver_success(zenith_simple_env: ZenithEnv):
         assert res["last_received_msg_lsn"] is not None, "the last received message's LSN is empty"
 
         last_msg_lsn = lsn_from_hex(res["last_received_msg_lsn"])
-        assert prev_msg_lsn is not None and prev_msg_lsn >= last_msg_lsn, \
+        assert prev_msg_lsn is None or prev_msg_lsn < last_msg_lsn, \
             f"the last received message's LSN {last_msg_lsn} hasn't been updated \
             compared to the previous message's LSN {prev_msg_lsn}"
 

